Lib/argparse.py

```diff
@@ -2556,8 +2554,8 @@ def _check_value(self, action, value):
         # converted value must be one of the choices (if specified)
         if action.choices is not None and value not in action.choices:
             args = {'value': value,
-                    'choices': ', '.join(map(repr, action.choices))}
-            msg = _('invalid choice: %(value)r (choose from %(choices)s)')
+                    'choices': ', '.join(map(str, action.choices))}
+            msg = _('invalid choice: %(value)s (choose from %(choices)s)')
```

Comment:
Actually, I think that it is better to keep `%(value)r` here.

* This is an invalid user input. It may be anything, including empty string, spaces or newline character -- raw string in an error message will be confusing.
* It does not break user translations.
* It does not break existing tests.

But we should first convert value to string. So I suggest to use `'value': str(value)` in a dict above.